src/tools/rbd/action/MirrorImage.cc

```diff
@@ -104,7 +128,13 @@ int execute_enable_disable(const po::variables_map &vm, bool enable,
     return r;
   }
 
-  r = enable ? image.mirror_image_enable() : image.mirror_image_disable(force);
+  librbd::mirror_image_mode_t mode = RBD_MIRROR_IMAGE_MODE_JOURNAL;
+  if (vm.count("mode")) {
```

Comment:
Does this work for positionals?

Reply:
Indeed. Thanks. We don't have a test  to catch this right now yet. I suppose we will have one in qa/workunits/rbd/rbd_mirror.sh when rbd-mirror supports snapshot mirroring.